airflow/providers/cncf/kubernetes/operators/custom_object_launcher.py

```diff
@@ -333,7 +333,7 @@ def spark_job_not_running(self, spark_obj_spec):
             except Exception:
                 pass
             raise AirflowException(f"Spark Job Failed. Error stack: {err}")
-        return driver_state == CustomObjectStatus.SUBMITTED
+        return driver_state != CustomObjectStatus.RUNNING
```

Comment:
can you add unit test to avoid regression?

Reply:
Okay, actually tests even now are failing. I think I need to mock `get_namespaced_custom_object_status` to return status of spark application with different states, maybe it can be like a generator, which will return statuses with states: pending, submitted, running

Reply:
Tests are fixed, new unit test for this case is added